src/scales/scale.ts

```diff
@@ -122,19 +118,23 @@ module Plottable {
       }
     }
 
-    protected _getDomain() {
-      return this._d3Scale.domain();
+    protected _getDomain(): D[] {
+      throw new Error("subclasses should override _getDomain");
     }
 
     protected _setDomain(values: D[]) {
       if (!this._domainModificationInProgress) {
         this._domainModificationInProgress = true;
-        this._d3Scale.domain(values);
+        this._setBackingDomain(values);
         this._dispatchUpdate();
         this._domainModificationInProgress = false;
       }
     }
 
+    protected _setBackingDomain(values: D[]) {
```

Comment:
I'll get to documentation after we get our API into a good state.  I'm mainly trying to focus on functionality and once we have the refactor into a somewhat stable state, I'll go document this.